pkg/microservices/microservices.go

```diff
@@ -0,0 +1,124 @@
+package microservices
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/hobbyfarm/gargantua/pkg/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type MicroService struct {
```

Comment:
Let's do `type MicroService string` here, then we can do..

```
const (
  MicroServiceAuthR MicroService = "authr-service"
  [etc]
)
```

Doing it this way will allow us to type the argument to createMicroService as a `MicroService`. Then calls to that would look like:
`createMicroService(MicroServiceAuthR)` instead of having to do a type check in that function.